mitodl/ccxcon
# Removed dredd, removed unused HTTP methods from API, added unit tests

#### What are the relevant tickets?

Fixes #121 
Fixes #83
#### What's this PR do?
- Removes dredd and related requirements like node.js
- Disables unused methods like PATCH, PUT, DELETE on course and module detail views, and POST on module list view.
- Adds unit tests to cover module and course APIs for GET methods.
#### Where should the reviewer start?

courses/views.py
#### How should this be manually tested?

Just run the unit tests locally. We might want to do this on a Mac as well to confirm #83 is fixed
